app/uk/gov/hmrc/incorporatedentityidentification/services/ValidateIncorporatedEntityDetailsService.scala

```diff
@@ -17,22 +17,26 @@
 package uk.gov.hmrc.incorporatedentityidentification.services
 
 import javax.inject.{Inject, Singleton}
-import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.{HeaderCarrier, NotFoundException}
 import uk.gov.hmrc.incorporatedentityidentification.connectors.GetCtReferenceConnector
-import uk.gov.hmrc.incorporatedentityidentification.models.{IncorporatedEntityDetailsValidationResult, DetailsMatched, DetailsMismatched, DetailsNotFound}
+import uk.gov.hmrc.incorporatedentityidentification.models.{DetailsDownstreamError, DetailsMatched, DetailsMismatched, DetailsNotFound, IncorporatedEntityDetailsValidationResult}
 
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class ValidateIncorporatedEntityDetailsService @Inject()(getCtReferenceConnector: GetCtReferenceConnector)(implicit ec: ExecutionContext) {
+class ValidateIncorporatedEntityDetailsService @Inject() (getCtReferenceConnector: GetCtReferenceConnector)(implicit ec: ExecutionContext) {
 
-  def validateDetails(companyNumber: String, optCtUtr: Option[String])(implicit hc: HeaderCarrier): Future[IncorporatedEntityDetailsValidationResult] = {
+  def validateDetails(companyNumber: String, optCtUtr: Option[String])(implicit
+    hc: HeaderCarrier
+  ): Future[IncorporatedEntityDetailsValidationResult] = {
     getCtReferenceConnector.getCtReference(companyNumber).map {
-      case Some(retrievedCtUtr) => optCtUtr match {
-        case Some(`retrievedCtUtr`) => DetailsMatched
-        case Some(_) | None => DetailsMismatched
-      }
-      case None => DetailsNotFound
+      case Right(retrievedCtUtr) =>
+        optCtUtr match {
+          case Some(`retrievedCtUtr`) => DetailsMatched
+          case Some(_) | None         => DetailsMismatched
+        }
```

Comment:
Would it be clearer to write lines 34-37 as follows?
```scala
if( optCtUtr contains retrievedCtUtr ) DetailsMatched
else                                   DetailsMismatched
```

Reply:
I agree, I did not want to change too much about the code but now there is two of us I'm happy to change it

Reply:
https://github.com/hmrc/incorporated-entity-identification/commit/fa036cd13df8b0feafbce960a22898ad5c4d823c